src/Tizen.System.Feedback/Feedback/Feedback.cs

```diff
@@ -250,24 +204,12 @@ public Feedback()
         public bool IsSupportedPattern(FeedbackType type, String pattern)
         {
             bool supported = false;
-            bool found = false;
-            int i = 0;
             Interop.Feedback.FeedbackError res;
 
-            for (i = 0; i < Constants.NumberOfPattern; i++)
-            {
-                if (String.Compare(pattern, Pattern[i].PatternString) == 0)
-                {
-                    found = true;
-                    break;
-                }
-            }
-
-            if (!found)
+            if (!Pattern.ContainsKey(pattern))
```

Comment:
Using ContainsKey and Item operator at the same time is redundant, because we use ContainsKey inside Items operator.
I recommend using TryGetValue.

```C#
if (Pattern.TryGetValue(pattern, out int value))
{
    res = res = (Interop.Feedback.FeedbackError)Interop.Feedback.IsSupportedPattern((Interop.Feedback.FeedbackType)type, value, out supported);
}
else throw new ArgumentException("Invalid Arguments");
```

Reply:
Thank you for point this, TryGetValue() seems more efficient way to access element of Dictionary. :)